pkg/controller/prpqr_reconciler/prpqr_reconciler.go

```diff
@@ -151,26 +151,50 @@ func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, logge
 		existingProwjobsByNameHash[pj.Labels[releaseJobNameLabel]] = &existingProwjobs.Items[i]
 	}
 
-	baseMetadata := metadataFromPullRequestUnderTest(prpqr.Spec.PullRequest)
-	prowjobs, errs := generateProwjobs(r.configResolverClient, r.prowConfigGetter.Config(), baseMetadata, req.Name, req.Namespace, prpqr.Spec.Jobs.Jobs, &prpqr.Spec.PullRequest)
-	for job, err := range errs {
-		createdJobs[job] = v1.PullRequestPayloadJobStatus{ReleaseJobName: job, Status: prowv1.ProwJobStatus{
-			State:       prowv1.ErrorState,
-			Description: err.Error(),
-		}}
+	statusByJobName := map[string]*v1.PullRequestPayloadJobStatus{}
+	for i := range prpqr.Status.Jobs {
```

Comment:
This will always be empty, so there is no need for this loop.

Reply:
That's not true. Status will have normal content e.g. when you restart the controller and that will make you re-reconcile all PRPQRs in the cluster.

Reply:
But overall, I think that the contrroller (any controller basically) should have as few assumptions about is the incoming state of the resource as possible. That makes it robust against changes. If you do not assume *anything* and always act only on what you actually observe, you are robust against any external change.